products/compute/api.yaml

```diff
@@ -7553,6 +7553,170 @@ objects:
         description: |
           The default port used if the port number is not specified in the
           network endpoint.
+  - !ruby/object:Api::Resource
+    name: 'RegionNetworkEndpointGroup'
+    kind: 'compute#networkEndpointGroup'
+    base_url: 'projects/{{project}}/regions/{{region}}/networkEndpointGroups'
+    input: true
+    min_version: beta
+    has_self_link: true
+    collection_url_key: 'items'
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/load-balancing/docs/negs/serverless-neg-concepts'
+      api: 'https://cloud.google.com/compute/docs/reference/rest/beta/regionNetworkEndpointGroups'
+    description: |
+      A regional NEG that can support Serverless Products.
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/regions/{{region}}/operations/{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'targetLink'
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'status'
+        complete: 'DONE'
+        allowed:
+          - 'PENDING'
+          - 'RUNNING'
+          - 'DONE'
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error/errors'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'region'
+        resource: 'Region'
+        imports: 'name'
+        description: |
+          A reference to the region where the Serverless NEGs Reside.
+        required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          Name of the resource; provided by the client when the resource is
+          created. The name must be 1-63 characters long, and comply with
+          RFC1035. Specifically, the name must be 1-63 characters long and match
+          the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means the
+          first character must be a lowercase letter, and all following
+          characters must be a dash, lowercase letter, or digit, except the last
+          character, which cannot be a dash.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          An optional description of this resource. Provide this property when
+          you create the resource.
+      - !ruby/object:Api::Type::Enum
+        name: 'networkEndpointType'
+        description: |
+          Type of network endpoints in this network endpoint group. Defaults to SERVERLESS
+        values:
+          - :SERVERLESS
+        default_value: :SERVERLESS
+      - !ruby/object:Api::Type::NestedObject
+        name: 'cloudRun'
+        exactly_one_of:
+          - cloudRun
+          - cloudFunction
+          - appEngine
+        at_least_one_of: 
```

Comment:
These need to be set on the fields 'service' and 'url_mask' themselves, otherwise TF will think that at least one of those fields needs to always be set, rather than just when 'cloud_run' is set.